structs.go

```diff
@@ -909,6 +916,59 @@ type GatewayBotResponse struct {
 	Shards int    `json:"shards"`
 }
 
+// GatewayStatusUpdate is sent by the client to indicate a presence or status update
+// https://discordapp.com/developers/docs/topics/gateway#update-status-gateway-status-update-structure
+type GatewayStatusUpdate struct {
+	Since  int      `json:"since"`
+	Game   Activity `json:"game"`
+	Status string   `json:"status"`
+	AFK    bool     `json:"afk"`
+}
+
+// Activity defines the Activity sent with GatewayStatusUpdate
+// https://discordapp.com/developers/docs/topics/gateway#activity-object
+type Activity struct {
+	Name string
+	Type ActivityType
+	URL  string
+}
+
+// ActivityType is the type of Activity (see ActivityType* consts) in the Activity struct
+// https://discordapp.com/developers/docs/topics/gateway#activity-object-activity-types
+type ActivityType int
+
+// Valid ActivityType values
+//https://discordapp.com/developers/docs/topics/gateway#activity-object-activity-types
+const (
+	ActivityTypeGame GameType = iota
+	ActivityTypeStreaming
+	ActivityTypeListening
+	//	ActivityTypeWatching // not valid in this use case?
+	ActivityTypeCustom = 4
```

Comment:
According to the documentation you provided, it does seem that `ActivityTypeWatching` is not a supported activity type. Would it be worth moving this line to above the commented out line to continue the `iota` without hard-coding it to `4`?

Reply:
Well, ActivityTypeCustom has to be a 4 (per Discord Docs) - so it has to be hard coded in there,  we can't just continue the iota because that line would get assigned a 3.  

The comment could possibly be removed but I liked it just for a bit of clarity.

Reply:
Ah good eye! I missed that 3 is skipped over in their doc. Probably a hint that will be supported in the future :)